### PR TITLE
BLD: make doc build work more robustly.

### PR DESCRIPTION
**Pull request by rgommers**
Follow-up to gh-12911

### CODE REVIEW

**Comment by mattip** [comment]
Is there a way we could clean up the Makefile by removing unused commands instead? It seems to me the `dist` variants are tied to a specific theoretical layout and not the job of vanilla numpy. Use of `easy-install` and eggs is also outdated, no?

But obviously I am wrong since there is a reason you are working on this ...

**Comment by rgommers** [comment]
>  It seems to me the `dist` variants are tied to a specific theoretical layout and not the job of vanilla numpy

no, `make dist` is the command to build the docs for releases, and that invokes `dist-build` and `real-dist`

> Use of `easy-install` and eggs is also outdated, no?

yeah, that could be changed. however, things work as is and testing makefile changes is slow, so I'd rather not touch it.

**Comment by charris** [comment]
Plain old `python` is Python 2.7 in the linux ecosystem, with `python3` for Python 3.*, and that convention will likely remain so for many years. As long as the build is Python 2.7 compatible and Python 2.7 is installed, that will probably be OK. Maybe we should try `python3` first with a fallback to `python`.

**Comment by rgommers** [comment]
> Maybe we should try `python3` first with a fallback to `python`.

I'd be surprised if any devs have it set up like that, because it'd be quite annoying to type `python3`, `pip3` etc. all the time. But can't hurt much either, so happy to change.

> As long as the build is Python 2.7 compatible

it's not, we've dropped 2.7

**Comment by charris** [comment]
Thanks Ralf.

**Comment by mattip** [comment]
This doesn't work. Makefiles [cannot handle colon](https://stackoverflow.com/questions/2052406/escaping-colons-in-filenames-in-a-makefile), so `make real-dist` fails. Fix on the way.

File: doc/Makefile [hunk 1]
```diff
@@ -3,8 +3,7 @@
 
 # This needs to be major.minor, just "3" doesn't work - it will result in
 # issues with the amendments to PYTHONPATH and install paths (see DIST_VARS).
-# If you're using a different Python version, just change the number here.
-PYVER = 3.6
+export PYVER=`python3 -c 'import sys; print("{0}.{1}".format(*sys.version_info[:2]))'`
```
Comment (eric-wieser) [inline doc/Makefile]: Isn't this shell syntax in a makefile? Does that work?

Comment (mattip) [reply doc/Makefile]: see #13063 for a fix

**Pull request merged**